src/main/java/edu/arizona/kra/negotiations/dao/ojb/NegotiationLookupDaoOjb.java

```diff
@@ -0,0 +1,284 @@
+     * @param addActivePendingVersion -> Boolean indicating if we need to look only for ACTIVE+PENDING versions of the associated BOs
+     *
+     * @return Criteria -> criteria for this lookup specific to the associated BO
+     */
+    private Criteria getCriteria(KraPersistableBusinessObjectBase bo, String associationType, String associatedId, Map<String,String> associatedValues, Map<String,String> transformKey, boolean addActivePendingVersion){
+        Map<String, String> filters = transformMap(associatedValues, transformKey);
+        if ( filters == null ){
+            LOG.debug("getCriteria: No additional filters for {}..", bo.getClass());
+            return new Criteria();
+        }
+        //creating subQuery for this particular's BO related filtering of negotiations
+        if ( addActivePendingVersion && bo instanceof InstitutionalProposal ) {
+            filters.put("proposalSequenceStatus", VersionStatus.ACTIVE.name());
+        }
+
+        Criteria associatedCriteria = getCollectionCriteriaFromMap(bo, filters);
+        //add active or pending VersionHistory criteria to the subquery
+        if ( addActivePendingVersion && !(bo instanceof InstitutionalProposal) ){
+                associatedCriteria.addAndCriteria(getActiveOrPendingVersionCriteria());
+        }
+        //create subquery for the associated object with the above criteria
+        ReportQueryByCriteria subQuery = QueryFactory.newReportQuery(bo.getClass(), associatedCriteria);
+        subQuery.setAttributes(new String[] {associatedId});
+
+        //create and return the negotiation search criteria including the above subquery
+        Criteria associatedBosCriteria = new Criteria();
+        associatedBosCriteria.addEqualTo(NEGOTIATION_TYPE_ATTR,
+                getNegotiationService().getNegotiationAssociationType(associationType).getId());
+        associatedBosCriteria.addIn(ASSOCIATED_DOC_ID_ATTR, subQuery);
+
+        LOG.debug("Exit getCriteria for class {} = {}", bo.getClass(), associatedBosCriteria );
+        return associatedBosCriteria;
+    }
+
+    /**
+     * Parses the user input for negotiation Age and returns the appropriate Criteria for it.
+     * If the input is empty or cannot be parsed, will return an empty Criteria.
+     *
+     * @param ageFilter Valid examples: integer, >minvalue, <maxvalue, minvalue..maxvalue
+     * @return
+     */
+    private Criteria getNegotiationAgeCriteria( String ageFilter ) {
+        if ( StringUtils.isEmpty(ageFilter) ) {
+            LOG.debug("getCriteria: No additional filters for ageFilter.");
+            return new Criteria();
+        }
+        Criteria negotiationAgeCriteria = new Criteria();
+        LOG.debug("Add negotiation age filter={}", ageFilter);
+
+        //Parse the ageFilter to find out what operation we need and what are the limits
+        //TODO refactor this old code...
```

Comment:
Nope, the code for parsing the criteria for the negotiation age still needs to be refactored...